lib/puppet/provider/package/yum.rb

```diff
@@ -151,6 +151,10 @@ def install
       end
     end
 
+    if @resource[:package_arch]
+      wanted = "#{wanted}.#{@resource[:package_arch]}"
+    end
+
```

Comment:
Did your update attempt to address uninstall? We also wonder about how doing install with a different package architecture would work if the package is already installed. The in_sync check will confirm the package is already installed, requiring you to explicitly uninstall it in one run, then install the other version in another.